lib/activity_notification/orm/active_record/notification.rb

```diff
@@ -45,7 +45,11 @@ class Notification < ::ActiveRecord::Base
         belongs_to :notifier,      polymorphic: true, optional: true
 
         # Serialize parameters Hash
-        serialize  :parameters, Hash
+        if Rails.gem_version >= Gem::Version.new('7.1')
+          serialize  :parameters, type: Hash, coder: YAML
```

Comment:
Based on reading the rails code and some tests, I believe YAML is the right selection